### PR TITLE
Re-enables missing gtest-1.10.0.patch

### LINKED ISSUE #2841: [package] gtest/1.10.0: missing patch
### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **gtest/1.10.0**
  * Operating System+version: **Linux Debian 10.4**
  * Compiler+version: **GCC 8**
  * Conan version: **conan 1.29.0**

### Conan profile
```
Configuration for profile default:

[settings]
os=Linux
os_build=Linux
arch=x86_64
arch_build=x86_64
compiler=gcc
compiler.version=8
compiler.libcxx=libstdc++11
build_type=Release
[options]
[build_requires]
[env]
CC=/usr/bin/gcc-8
CXX=/usr/bin/g++-8
```

### Steps to reproduce (Include if Applicable)

Just retrieve/build the package with `gtest:debug_postfix="something"`. The value is not used.

### My guess
The `gtest-1.10.0.patch` is not applied but I think it should be.

Seems problem is due to the [conandata.yml file](https://github.com/conan-io/conan-center-index/blob/master/recipes/gtest/all/conandata.yml) :
```yml
patches:
  "1.10.0":
    - patch_file: "patches/gtest-1.10.0.patch"
      base_path: "source_subfolder"
  "1.10.0":
    - patch_file: "patches/gtest-1.10.0-override.patch"
      base_path: "source_subfolder"
```

should be written : 
```yml
patches:
  "1.10.0":
    - patch_file: "patches/gtest-1.10.0.patch"
      base_path: "source_subfolder"
    - patch_file: "patches/gtest-1.10.0-override.patch"
      base_path: "source_subfolder"
```
or am I wrong ?

### PR DESCRIPTION
**Pull request by oleurodecision**
Specify library name and version:  **lib/1.0**

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

Fixes #2841

### CODE REVIEW

**Comment by oleurodecision** [comment]
Think it would also be good to move the `debug_postfix` deletion from the config_options step to the configure step, that is to say :

```python3
    def config_options(self):
        if self.settings.os == "Windows":
            del self.options.fPIC
        if self.settings.build_type != "Debug":
            del self.options.debug_postfix

    def configure(self):
        if self.settings.os == "Windows":
            if self.settings.compiler == "Visual Studio" and Version(self.settings.compiler.version.value) <= "12":
                raise ConanInvalidConfiguration("Google Test {} does not support Visual Studio <= 12".format(self.version))
```
to
```python3
    def config_options(self):
        if self.settings.os == "Windows":
            del self.options.fPIC

    def configure(self):
        if self.settings.os == "Windows":
            if self.settings.compiler == "Visual Studio" and Version(self.settings.compiler.version.value) <= "12":
                raise ConanInvalidConfiguration("Google Test {} does not support Visual Studio <= 12".format(self.version))
        if self.settings.build_type != "Debug":
            del self.options.debug_postfix
```

Do you agree ? Should it be done as a separate PR ?

**Comment by Croydon** [comment]
`config_options()` is the right place to conditionally delete options. What were your thoughts on moving it to `configure()`?

**Comment by oleurodecision** [comment]
well, basically because it allows using the same conanfile.txt for release and debug build types, when sometimes the option needs to be set (for some consumoing client projects)
Imho, deleting this kind of option that do not change the binary output (only the library name here, not like PIC option for instance) is quite overkilling, thus the question.

**Comment by uilianries** [comment]
`configure` is used most for deleting option when you need to check an option first, which is not this case. The `config_options` is specific for deleting options when you need to check `settings` first, just like the current situation. Please, keep it as is.

**Review by uilianries — APPROVED** [review]

**Review by Croydon — APPROVED** [review]

File: recipes/gtest/all/conandata.yml [hunk 1]
```diff
@@ -12,6 +12,5 @@ patches:
   "1.10.0":
     - patch_file: "patches/gtest-1.10.0.patch"
       base_path: "source_subfolder"
-  "1.10.0":
```
Comment (SSE4) [inline recipes/gtest/all/conandata.yml:15]: this is interesting typo, @uilianries do you think if it's something that can be easily validated by hook?

**Review by SSE4 — APPROVED** [review]

File: recipes/gtest/all/conandata.yml (hunk 1 above)
Comment (uilianries) [reply recipes/gtest/all/conandata.yml:15]: @SSE4 Yes, I think so.

**Review by intelligide — APPROVED** [review]

**Review by danimtb — APPROVED** [review]

**Pull request merged**